Feature/user role 37093795

Bukarinov opened the pull request:
https://www.pivotaltracker.com/story/show/37093795

Bukarinov committed:
Rename parent to parents.

Bukarinov committed:
Fix misprint

Bukarinov committed:
Fix misprint

Bukarinov committed:
Fix misprint

Bukarinov committed:
Rename parent to parents - fix test

Bukarinov committed:
Remove ZF ACL.
Create Assertion draft.

Bukarinov committed:
Remove test for Acl.

Bukarinov committed:
Move Grants from library to application

Bukarinov committed:
Merge master

shumkov commented on library/GeometriaLab/Permissions/Assertion/Assertion.php:
```diff
@@ -0,0 +1,126 @@
+     */
+    public function addResource(ResourceInterface $resource)
+    {
+        $resourceId = $resource->getId();
```
getName

Bukarinov committed:
Rename Resource->id to Resource->name

Bukarinov committed:
GeometriaLab\Model\Schemaless\Model instead of GeometriaLab\Model\AbstractModel

Bukarinov committed:
Custom collection for Grant

shumkov commented on library/GeometriaLab/Permissions/Assertion/Assertion.php:
```diff
@@ -0,0 +1,126 @@
+     * @param AbstractModel $params
+     * @return bool
+     */
+    public function assert($resource, $privilege, AbstractModel $params = null)
```
$resource, $privilege, $arg1 = null, $arg2 = null, $argN = null

shumkov commented on library/GeometriaLab/Permissions/Assertion/Assertion.php:
```diff
@@ -0,0 +1,126 @@
+     */
+    public function assert($resource, $privilege, AbstractModel $params = null)
+    {
+        if (!$this->hasResource($resource)) {
```
http://php.net/manual/en/function.func-get-args.php

shumkov commented on library/GeometriaLab/Permissions/Assertion/Assertion.php:
```diff
@@ -0,0 +1,126 @@
+     */
+    public function hasResource($resource)
+    {
+        $resourceId = (string) $resource;
```
если инстанс интерфеса то getName

shumkov commented on library/GeometriaLab/Permissions/Assertion/Resource.php:
```diff
@@ -0,0 +1,143 @@
+        return isset($this->privileges[$privilege]);
+    }
+
+    /**
```
remove

shumkov commented on library/GeometriaLab/Permissions/Assertion/Resource.php:
```diff
@@ -0,0 +1,143 @@
+     * @return bool
+     * @throws \InvalidArgumentException
+     */
+    public final function assert(Assertion $assertion, $privilege, Model $model = null)
```
эту логику в assertion

shumkov commented on library/GeometriaLab/Permissions/Assertion/Resource.php:
```diff
@@ -0,0 +1,143 @@
+        return $this->privileges;
+    }
+
+    /**
```
это не надо

shumkov commented on library/GeometriaLab/Permissions/Assertion/ResourceInterface.php:
```diff
@@ -0,0 +1,33 @@
+     */
+    public function getName();
+
+    /**
```
убрать

shumkov commented on library/GeometriaLab/Permissions/Assertion/ServiceFactory.php:
```diff
@@ -1,101 +1,96 @@
-    Zend\Permissions\Acl\Role\GenericRole as ZendGenericRole;
+    Zend\ServiceManager\ServiceLocatorInterface as ZendServiceLocatorInterface;
 
 class ServiceFactory implements ZendFactoryInterface
```
помойму это не фабрика

Bukarinov committed:
Rename ServiceFactory to Service.
Simple AbstractResource and ResourceInterface.
Pass all parameters from assertion to resource.

Bukarinov committed:
Move Guest, Permission, Permissions, User and UserInterface to Model/User.
Rename grant to permissions.

Bukarinov committed:
Add array of privileges which always allowed for all.

Bukarinov committed:
Tests for Assertion

shumkov commented on library/GeometriaLab/Permissions/Assertion/AbstractResource.php:
```diff
@@ -0,0 +1,47 @@
+     *
+     * @return array
+     */
+    public function getAllowedPrivileges()
```
---

shumkov commented on library/GeometriaLab/Permissions/Assertion/Assertion.php:
```diff
@@ -0,0 +1,139 @@
+    {
+        $resource = $this->getResource($resource);
+
+        if (in_array($privilege, $resource->getAllowedPrivileges())) {
```
---

shumkov commented on library/GeometriaLab/Permissions/Assertion/Service.php:
```diff
@@ -0,0 +1,141 @@
+
+namespace GeometriaLab\Permissions\Assertion;
+
+use GeometriaLab\Api\Mvc\Router\Http\Api;
```
---

Bukarinov committed:
Remove allowedPrivileges from Resource.
